term/fhir-term/src/main/java/com/ibm/fhir/term/service/FHIRTermService.java

```diff
@@ -1021,38 +1079,55 @@ private Uri getSource(ConceptMap conceptMap) {
-            java.lang.String system = null;
-            if (coding.getSystem() != null) {
-                system = coding.getSystem().getValue();
-            } else if (codeSystem != null && codeSystem.getUrl() != null) {
-                system = codeSystem.getUrl().getValue();
-            }
-            boolean caseSensitive = (codeSystem != null) ? CodeSystemSupport.isCaseSensitive(codeSystem) : false;
-            if (codeSystem == null && system != null) {
-                java.lang.String version = (coding.getVersion() != null) ? coding.getVersion().getValue() : null;
-                java.lang.String url = (version != null) ? system + "|" + version : system;
-                caseSensitive = CodeSystemSupport.isCaseSensitive(url);
-            }
-            result = caseSensitive ? outcome.getDisplay().equals(coding.getDisplay()) : normalize(outcome.getDisplay().getValue()).equals(normalize(coding.getDisplay().getValue()));
-            message = !result ? java.lang.String.format("The display '%s' is incorrect for code '%s' from code system '%s'", coding.getDisplay().getValue(), coding.getCode().getValue(), system) : null;
+
+        java.lang.String system = null;
+        if (coding.getSystem() != null) {
+            system = coding.getSystem().getValue();
+        } else if (codeSystem != null && codeSystem.getUrl() != null) {
+            system = codeSystem.getUrl().getValue();
         }
+        boolean caseSensitive = (codeSystem != null) ? CodeSystemSupport.isCaseSensitive(codeSystem) : false;
+        if (codeSystem == null && system != null) {
+            java.lang.String version = (coding.getVersion() != null) ? coding.getVersion().getValue() : null;
+            java.lang.String url = (version != null) ? system + "|" + version : system;
+            caseSensitive = CodeSystemSupport.isCaseSensitive(url);
+        }
+        boolean result = caseSensitive ? lookupOutcome.getDisplay().equals(coding.getDisplay()) : 
+                normalize(lookupOutcome.getDisplay().getValue()).equals(normalize(coding.getDisplay().getValue()));
+        java.lang.String message = !result ? java.lang.String.format("The display '%s' is incorrect for code '%s' from code system '%s'", 
+                coding.getDisplay().getValue(), coding.getCode().getValue(), system) : null;
+        
         return ValidationOutcome.builder()
                 .result(result ? Boolean.TRUE : Boolean.FALSE)
                 .message((message != null) ? string(message) : null)
-                .display((outcome != null) ? outcome.getDisplay() : null)
+                .display(lookupOutcome.getDisplay())
                 .build();
     }
 
-    private ValidationOutcome validateCode(Coding coding, boolean result, LookupOutcome outcome) {
-        return validateCode(null, coding, result, outcome);
+    private ValidationOutcome buildNotFoundValidationOutcome(Coding coding) {
```

Comment:
Looks like this method is only called once. Maybe get rid of it and go with the generic buildValidationOutcome() method, and where this is currently called follow your pattern in other cases of building the message then building the validation outcome (which would now be via the generic buildValidationOutcome() method).

Reply:
done